packages/boot/test/bootstrapTests/benchmarkerator.md

```diff
@@ -0,0 +1,171 @@
+(which will be described in detail below)
+
+After defining one or more benchmarks with `addBenchmark` you must then end by
+invoking:
+
+```await bench.run(name);```
+
+where `name` is a string naming the benchmark run for purposes of error and
+result reporting.
+
+#### The `Benchmark` object
+
+The `benchmark` object may have the following properties, all of which are
+optional except for `executeRound`:
+
+`setup?: (context: BenchmarkContext) => Promise<Record<string, unknown> | undefined>`
+
+  An optional async method to perform any pre-run setup that you need to do. It
+  may optionally return a configuration record containing any benchmark-specific
+  information you desire.  This will be provided as part of the context object
+  in subsequent calls into the benchmark.
+
+`executeRound: (context: BenchmarkContext, round: number) => Promise<void>`
+
+  A required async method which is called to actually execute one round of the
+  benchmark.  The `round` argument is the number of the benchmark round
+  that this call to `executeRound` is being asked to execute.
+
+`finish?: (context: BenchmarkContext) => Promise<void>`
+
+  An optional async method to perform any post-run teardown that you need to do.
+
+`rounds?: number`
+
+  The number of benchmark rounds that will be run if not overridden on the
+  command line.  If omitted, it defaults to `1`.
+
+#### The `BenchmarkContext` object
+
+The first parameter to each of the benchmark methods is a context object that
+provides various information about the execution.  It has the properties:
+
+`options: Record<string, string>`
+
+  Named options from the command line (see Command Line below).
+
+`argv: string[]`
+
+  Remaining unparsed arguments from the command line.
+
+`actors: Record<string, SmartWalletDriver>`
+
+  Wallet drivers for the various personae that exist in the test environment.
+  Currently present are `gov1`, `gov2`, and `gov3` (the governance committee);
+  `atom1` and `atom2` (ersatz ATOM-USD price feed oracles); and `alice`, `bob`,
+  and `carol` (arbitrary users to participate in interactions being exercised).
+
+`label: string`
+
+  The benchmark label, from the `addBenchmark` call that defined this benchmark.
+
+`rounds: number`
+
+  The total number of benchmark rounds that are to be executed in this run.
+
+`config?: Record<string, unknown>`
+
+  The configuration object that was returned by the `setup` method, if one was.
+
+## Executing a benchmark
+
+The benchmark that you define by incorporating The Benchmarkerator is a
+standalone Node executable.  You run it with the command:
+
+`node benchmark-yourbenchmarkname.js [OPTIONS...]`
+
+The supported command line options are:
+
+| Option | What |
+|--------|------|
+| `-r N`<br/>`--rounds N` | Execute _N_ rounds of each benchmark |
+| `-b PATT`<br/>`--benchmark PATT` | Only execute benchmarks matching _PATT_ (may be specified more than once; this is similar to Ava's `-m` option)|
```

Comment:
This is one of the cases where I don't think it pays to get fancy.  We should either just picking globbing or pick regexes and leave it at that.  I went with regex mainly because it was easy.  I've actually been operating under the belief that Ava `-m` does regex matching too -- is that mistaken?  I guess globbing, "`*`" matches anything, which is the most common thing to want to match in cases like this, whereas with regex you have to write something like "`.*`" which is probably a pain.  OK, I've talked myself into globbing.  npmjs.org lists like a gajillion globbing packages.  Do we have a preferred one?